DataStore2/init.lua

```diff
@@ -179,13 +179,34 @@ function DataStore:GetTableAsync(default, ...)
 	end)
 end
 
+function DataStore:SetValidator(validator)
+	assert(
+		type(validator) == "function",
+		"function expected, got "..type(validator)
+	)
+	self.validator = validator
+end
+
 function DataStore:Set(value, _dontCallOnUpdate)
+	if self.validator then
+		assert(
+			self.validator(value),
+			"Invalid data"
```

Comment:
Same comment as what's up ahead.

Reply:
This still needs a more detailed error message, *and* should probably be different between :Set and :Update so that it's more obvious which function failed.